GNUmakefile

```diff
@@ -4,155 +4,49 @@ VERSION = devel
+.PHONY: binaries
+binaries: dkcheck.native dktop.native dkdep.native dkprune.native
 
-KERNEL_MLI := $(wildcard kernel/*.mli)
-KERNEL_ML  := $(KERNEL_MLI:.mli=.ml)
+dkdep.native:
+	@ln -fs _build/install/default/bin/dkdep dkdep.native
 
-.PHONY: kernel
-kernel: _build/kernel/kernel.cma _build/kernel/kernel.cmxa
+dkcheck.native:
+	@ln -fs _build/install/default/bin/dkcheck dkcheck.native
 
-kernel/version.ml: GNUmakefile
-	@echo "[GEN] $@ ($(VERSION))"
-	$(Q)echo 'let version = "$(VERSION)"' > $@
-
-_build/kernel/kernel.cma: $(KERNEL_MLI) $(KERNEL_ML)
-	@echo "[BYT] $@"
-	$(Q)ocamlbuild -quiet -use-ocamlfind kernel/kernel.cma
-
-_build/kernel/kernel.cmxa: $(KERNEL_MLI) $(KERNEL_ML)
-	@echo "[OPT] $@"
-	$(Q)ocamlbuild -quiet -use-ocamlfind kernel/kernel.cmxa
-
-#### Compilation of the API library #######################################
-
-API_MLI := $(wildcard api/*.mli)
-API_ML  := $(API_MLI:.mli=.ml)
-
-.PHONY: api
-api: kernel _build/api/api.cma _build/api/api.cmxa
-
-_build/api/api.cma: $(API_MLI) $(API_ML)
-	@echo "[BYT] $@"
-	$(Q)ocamlbuild -quiet -use-ocamlfind api/api.cma
-
-_build/api/api.cmxa: $(API_MLI) $(API_ML)
-	@echo "[OPT] $@"
-	$(Q)ocamlbuild -quiet -use-ocamlfind api/api.cmxa
-
-#### Compilation of the parser library #######################################
-
-PARSER_MLI := $(wildcard parser/*.mli)
-PARSER_ML  := $(PARSER_MLI:.mli=.ml)
-PARSER_GEN := parser/menhir_parser.mly parser/lexer.mll
-
-.PHONY: parser
-parser: kernel _build/parser/parser.cma _build/parser/parser.cmxa
-
-_build/parser/parser.cma: $(PARSER_MLI) $(PARSER_ML) $(PARSER_GEN)
-	@echo "[BYT] $@"
-	$(Q)ocamlbuild -quiet -use-ocamlfind parser/parser.cma
-
-_build/parser/parser.cmxa: $(PARSER_MLI) $(PARSER_ML) $(PARSER_GEN)
-	@echo "[OPT] $@"
-	$(Q)ocamlbuild -quiet -use-ocamlfind parser/parser.cmxa
+dktop.native:
+	@ln -fs _build/install/default/bin/dktop dktop.native
 
-#### Compilation of the dedukti suite ########################################
+dkprune.native:
+	@ln -fs _build/install/default/bin/dkprune dkprune.native
 
-.PHONY: commands
-commands: dkcheck.native dkdep.native dktop.native dkprune.native
+.PHONY: bin
+bin: kernel/version.ml
+	@dune build
 
-dkcheck.native: kernel api parser commands/dkcheck.ml
-	@echo "[OPT] $@"
-	$(Q)ocamlbuild -quiet -use-ocamlfind commands/dkcheck.native
-
-dkdep.native: kernel api parser commands/dkdep.ml
-	@echo "[OPT] $@"
-	$(Q)ocamlbuild -quiet -use-ocamlfind commands/dkdep.native
-
-dktop.native: kernel api parser commands/dktop.ml
-	@echo "[OPT] $@"
-	$(Q)ocamlbuild -quiet -use-ocamlfind commands/dktop.native
-
-dkprune.native: kernel api parser commands/dkprune.ml
-	@echo "[OPT] $@"
-	$(Q)ocamlbuild -quiet -use-ocamlfind commands/dkprune.native
+.PHONY: doc
+doc:
+	@dune build @doc
 
-#### Generation of the documentation #########################################
+.PHONY: clean
+clean:
+	@dune clean
+	@rm -f dkcheck.native dktop.native dkdep.native dkprune.native
```

Comment:
Ce sont des liens symboliques, pourquoi les suppriner ?

Reply:
They become dangling links (`_build` folder deleted) once the project is cleaned, why keep them ?